Backend/Controllers/ProductController.cs

```diff
@@ -0,0 +1,229 @@
+﻿using Data;
+using Data.Models.ProductTables;
+using Backend.Services;
+using Lib.EntityFrameworkCore;
+using Microsoft.AspNetCore.Mvc;
+using Lib.Storage;
+using Lib.Storage.Services;
+using System.Drawing;
+using System.Drawing.Imaging;
+using Microsoft.EntityFrameworkCore;
+
+namespace Backend.Controllers
+{
+    [Route("v1/product")]
+    [ApiController]
+    public class ProductController : ControllerBase
+    {
+        private readonly Db db;
+
+        private readonly ImageService imageService;
+
+        private readonly Storj storage;
+
+
+        public ProductController(Db context, ImageService imageService, Storj storage)
+        {
+            db = context;
+            this.imageService = imageService;
+            this.storage = storage;
+        }
+
+
+        [HttpPost]
+        public async Task<IActionResult> CreateProduct(Product newProduct)
```

Comment:
You don't get Product from create request. At first request, you create a new product from nothing. Maximum you get slug or name.